source/service_test.go

```diff
@@ -267,13 +281,32 @@ func testServiceEndpoints(t *testing.T) {
 			},
 		},
 		{
+			"services annotated with legacy molecule annotations return an endpoint in compatibility mode",
+			"",
+			"testing",
+			"foo",
+			"molecule",
+			"",
+			map[string]string{
+				"dns": "route53",
+			},
+			map[string]string{
+				"domainName": "foo.example.org.",
```

Comment:
Does this work without the trailing `.` as well? I think that's the most common use case for us. But this format is supported too.